Sources/ScoutCLT/Main/ScoutCommand.swift

```diff
@@ -4,42 +4,92 @@
 // MIT license, see LICENSE file for details
 
 import Foundation
-import ArgumentParser
 import Scout
+import ArgumentParser
 import Lux
 
-private let abstract =
-"""
-Read and modify values in specific format file or data. Currently supported: Json, Plist and Xml.
-"""
+protocol ScoutCommand: ParsableCommand {
+
+    var inputFilePath: String? { get }
+
+    /// Called with the correct `PathExplorer` when `inferPathExplorer(from:in:)` completes
+    func inferred<P: PathExplorer>(pathExplorer: P) throws
+}
+
+extension ScoutCommand {
+
+    func run() throws {
+        let data = try readDataOrInputStream(from: inputFilePath)
+        try inferPathExplorer(from: data, in: inputFilePath)
+    }
+
+    /// Try to read data from the optional `filePath`. Otherwise, return the data from the standard input stream
+    func readDataOrInputStream(from filePath: String?) throws -> Data {
+        if let filePath = filePath {
+            return try Data(contentsOf: URL(fileURLWithPath: filePath.replacingTilde))
+        } else {
+            return FileHandle.standardInput.readDataToEndOfFile()
+        }
+    }
+
+    func inferPathExplorer(from data: Data, in inputFilePath: String?) throws {
+
+        if let json = try? Json(data: data) {
```

Comment:
Just for my info, would it be possible to replace those with a `switch (true) { ... }`?

Reply:
I don't think so since the `PathExplorer` has a Self requirement. But I would gladly be wrong.